### ISSUE TITLE
Show initialization error for user-defined roles in GUI

### ISSUE THREAD

**Issue by sharonovd** [tarantool/cartridge#45]
In GitLab by @opomuc on Jan 15, 2019, 12:31

Currently it marks instance "red" and it is impossible to do anything. It will be nice to have error message shown right away, instead of grepping logs

replicator.lua
```
local function init(opts)
   ...
   error('Some error message')
   ...
end
```

![image](/uploads/18df3c34b26660d5b58bfd94c23e465a/image.png)

At the same time, using graphql you will get
```
curl -X POST http://localhost:8081/graphql -d@- <<'QUERY'
{"query":
    "mutation {
        p1: probe_server(uri:\"tnt-router:3301\")
        p2: probe_server(uri:\"tnt-storage_1:3301\")
        p3: probe_server(uri:\"tnt-storage_2:3302\")

        j1: join_server(
            uri:\"tnt-router:3301\",
            instance_uuid: \"aaaaaaaa-aaaa-4000-b000-000000000001\",
            replicaset_uuid: \"aaaaaaaa-0000-4000-b000-000000000000\",
            roles: [\"vshard-router\", \"router\"]
        )
        j2: join_server(
            uri:\"tnt-storage_1:3301\",
            instance_uuid: \"bbbbbbbb-bbbb-4000-b000-000000000001\",
            replicaset_uuid: \"bbbbbbbb-0000-4000-b000-000000000000\",
            roles: [\"vshard-storage\",\"storage\"],
            timeout: 5
        )
        j3: join_server(
            uri:\"tnt-storage_2:3302\",
            instance_uuid: \"cccccccc-cccc-4000-b000-000000000001\",
            replicaset_uuid: \"cccccccc-0000-4000-b000-000000000000\",
            roles: [\"vshard-storage\",\"storage\"],
            timeout: 5
        )

        bootstrap_vshard

        cluster {
            failover(enabled:true)
        }

        e3: edit_replicaset(
            uuid: \"aaaaaaaa-0000-4000-b000-000000000000\"
            master: \"aaaaaaaa-aaaa-4000-b000-000000000001\"
            roles: [\"vshard-router\",\"router\",\"replicator\"]
        )
    }"
}
QUERY

{"errors":[{"message":".\/replicator.lua:36: Some error message"}]}
```

**Comment by sharonovd** [comment]
In GitLab by @rosik on Jan 15, 2019, 13:34

marked this issue as related to #32

**Comment by sharonovd** [comment]
In GitLab by @opomuc on Jan 15, 2019, 13:44

changed title from **Show initialization error in GUI** to **Show initialization error {+for user-defined roles +}in GUI**

**Comment by sharonovd** [comment]
In GitLab by @opomuc on Jan 15, 2019, 13:45

changed the description

**Comment by sharonovd** [comment]
In GitLab by @opomuc on Jan 15, 2019, 13:50

changed the description

**Comment by sharonovd** [comment]
In GitLab by @opomuc on Jan 15, 2019, 13:53

changed the description

**Comment by sharonovd** [comment]
In GitLab by @rosik on Jul 18, 2019, 19:42

This isformation isn't available in API, so I removed 'ui' label

**Comment by rosik** [comment]
To be fixed together with #397

**Comment by rosik** [comment]
Let's implement the followring GraphQL API and utilize it in the "server details" dialog:

```graphql
query{
  servers {
    boxinfo {
      cartridge {
        state
        error
      }
    }
  } 
}
```

See also #489
